lib/internal/readline/promises.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const {
+  ArrayPrototypeJoin,
+  ArrayPrototypePush,
+  Promise,
+  PromiseResolve,
+} = primordials;
+
+const { CSI } = require('internal/readline/utils');
+const { validateBoolean, validateInteger } = require('internal/validators');
+const { isWritable } = require('internal/streams/utils');
+const { codes: { ERR_INVALID_ARG_TYPE } } = require('internal/errors');
+
+const {
+  kClearToLineBeginning,
+  kClearToLineEnd,
+  kClearLine,
+  kClearScreenDown,
+} = CSI;
+
+class Readline {
+  #autoCommit = false;
+  #stream;
+  #todo = [];
```

Comment:
not blocking but private fields are still very slow in V8. Hidden symbols would probably be a lot more efficient (and consistent with what we do elsewhere)